src/app/index.ts

```diff
@@ -180,8 +183,47 @@ async function startApp() {
     } else if (runMode === 'web') {
         // Run WebUI with configured MCP identity (pass agentCard only)
         const agentCard = services.configManager.getConfig().agentCard ?? {};
-        startWebUI(agent, webPort, agentCard);
-        logger.info(`WebUI available at http://localhost:${webPort}`, null, 'magenta');
+        const frontPort = webPort;
+        const apiPort = webPort + 1;
+        const nextCwd = path.resolve(process.cwd(), 'src', 'app', 'webui');
+        logger.info(
+            `Starting Next.js dev server on http://localhost:${frontPort}`,
+            null,
+            'cyanBright'
+        );
+        const nextProc = spawn('npm', ['run', 'dev', '--', '--port', String(frontPort)], {
+            cwd: nextCwd,
+            shell: true,
+            stdio: 'inherit',
+            env: {
+                ...process.env,
+                NODE_ENV: 'development',
+                API_PORT: String(apiPort),
+                NEXT_PUBLIC_WS_URL:
+                    process.env.NEXT_PUBLIC_WS_URL ??
+                    (() => {
+                        const interfaces = os.networkInterfaces();
+                        for (const list of Object.values(interfaces)) {
+                            for (const iface of list ?? []) {
+                                if (iface.family === 'IPv4' && !iface.internal) {
+                                    return `ws://${iface.address}:${apiPort}`;
+                                }
+                            }
+                        }
+                        return `ws://localhost:${apiPort}`;
+                    })(),
+            },
+        });
+        nextProc.on('error', (err) => {
+            logger.error('Failed to start Next.js process:', err);
+        });
+        nextProc.on('exit', (code, signal) => {
+            logger.info(`Next.js process exited with code ${code} and signal ${signal}`);
+        });
+        // Start Express API server (for WebSocket and HTTP endpoints)
+        startServer(agent, apiPort, agentCard);
+        logger.info(`API endpoints available at http://localhost:${apiPort}`, null, 'magenta');
+        logger.info(`Frontend available at http://localhost:${frontPort}`, null, 'magenta');
```

Comment:
we aren't consistently sticking to a convention of env variable or localhost

this will be confusing if `process.env.NEXT_PUBLIC_WS_URL` is passed - it will still log localhost

similar for the `process.env.API_PORT` we added

we should fix this - essentially standardize on a few env variables controlling express and nextjs servers and then use them everywhere properly. ok to do this in a future PR as well but yeah we should do it